Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,174 @@
+- Include parameter options to select Stable, Preview, or LTS releases
+
+## Goals/Non-Goals
+
+Goals:
+
+- By default, download and install the latest LTS release to the users CurrentUser Scope
+  - CurrentUser - default installation scope location TODO:$env:LOCALAPPDATA\Microsoft\PowerShell. TODO: If the path
+    exists, remove and replace.
+  - AllUsers - Administrative users may explicit select this scope. AllUsers requires administrative
+    elevation and performs a silent install using the MSI defaults.
+- Allow advanced users to specify channel Stable/Preview/LTS
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
+- User selected versions prior to current (No version parameter)
+- Disconnected scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+- Cross platform support
+- Deployments at scale
+
+## Specification
+
+Command name: install-powershell7
+
+The default behavior is to install the zip package silently to CurrentUser scope, with the latest LTS
+release of PowerShell. If the user has administrative
+privilege, they can choose to install the msi package silently to the scope of AllUsers.
+
+- The stable package install location - $env:LOCALAPPDATA\Microsoft\PowerShell-Stable
+  - If path exists, detect installed version and compare:
+    - If versions equal: Provide an error "PowerShell 7 is already installed" and exit.
+    - If versions not equal: Overwrite original installation.
+- The preview package install location - $env:LOCALAPPDATA\Microsoft\PowerShell-Preview
+  - If path exists, detect installed version and compare:
+    - If versions equal: Provide an error "PowerShell 7 is already installed" and exit.
+    - If versions not equal: Overwrite original installation.
+- The installer should update the Users PATH for pwsh
+- The installer should enable Microsoft Updates for scope AllUsers.
+- The command should show a progress bar during the download and installation of PowerShell 7.
+- The command should produce an error if the download location (US based url) fails.
+
+The following parameters can be added by experienced PowerShell users to customize the installation.
+
+- Parameter: Scope - overrides the default of CurrentUser and allows the user to specify the
+  installation location AllUsers.
+- Parameter: Channel - options Stable/Preview/LTS - allows the user to select if they want the
+  current preview or LTS release.
+
+## Syntax
+
+The command `install-powershell7` is lower case and supports `-` or `--` parameters to be consistent
+with other command-line tools.
+
+```syntax
+install-powershell7  [--channel <option>] [--scope <option>] [--force]
+install-powershell7  [-c <option>] [-s <option>] [-f]
+```
+
+The `--channel <option>` must be one of the following:
+
+- `--channel preview`
```

Comment:
Maybe it's worth writing to the RFC that for preview installations, the path to be put in $env:PATH should be `<path-to-preview-folder>\preview`. That's where the `pwsh-preview.cmd` command locates. We need to do this so that the user can differentiate between the stable installation and preview installation by running `pwsh` vs. `pwsh-preview`.